cpp/src/arrow/flight/client.cc

```diff
@@ -385,9 +385,6 @@ class GrpcStreamWriter : public FlightStreamWriter {
     }
     return Status::OK();
   }
-  void set_memory_pool(MemoryPool* pool) override {
-    batch_writer_->set_memory_pool(pool);
-  }
```

Comment:
Should we add `const IpcWriteOptions& options` to `GrpcStreamWriter::Open()` instead?

Reply:
Yes, we should do that. It will have to happen after @lidavidm's refactoring goes in. Opening a JIRA https://issues.apache.org/jira/browse/ARROW-8190